master/internal/db/postgres_experiments.go

```diff
@@ -497,12 +501,16 @@ FROM trials t
 			"failed to get validation metrics for hyperparameter importance")
 	}
 	for _, row := range rows {
-		result, err := unmarshalHPImportanceHParams(row)
+		result, batches, err := unmarshalHPImportanceHParams(row)
 		if err != nil {
 			return nil, errors.Wrap(err,
 				"Failed to process validation metrics for hyperparameter importance")
 		}
-		results = append(results, result)
+		if previous, exists := results[batches]; exists {
+			results[batches] = append(previous, result)
+		} else {
+			results[batches] = []model.HPImportanceTrialData{result}
+		}
```

Comment:
```suggestion
		results[batches] = append(results[batches], result)
```
go should handle this just fine e.g. https://play.golang.org/p/6xaMh0qJ8SJ

Reply:
Already done in my fork. I think gofmt -s actually did this for me, which impresses me..